main/src/ca/uwaterloo/flix/tools/pkg/FlixPackageManager.scala

```diff
@@ -17,43 +17,76 @@ package ca.uwaterloo.flix.tools.pkg
 
 import ca.uwaterloo.flix.tools.Packager
 import ca.uwaterloo.flix.tools.Packager.getLibraryDirectory
+import ca.uwaterloo.flix.tools.pkg.Dependency.FlixDependency
 import ca.uwaterloo.flix.tools.pkg.github.GitHub
-import ca.uwaterloo.flix.util.Result.ToOk
+import ca.uwaterloo.flix.util.Result.{Err, Ok, ToOk}
 import ca.uwaterloo.flix.util.{Options, Result}
 
-import java.io.File
+import java.io.{File, PrintStream}
 import java.nio.file.{Files, Path, StandardCopyOption}
 import scala.util.Using
 
 object FlixPackageManager {
 
   // TODO: Move functionality from "Packager" in here.
 
+  //TODO: report errors
+  //TODO: tests
+  //TODO: comments
+  def installAll(manifest: Manifest, path: Path)(implicit out: PrintStream): Result[Unit, PackageError] = {
+    val flixDeps = findFlixDependencies(manifest)
+    for(dep <- flixDeps) {
+      val depName: String = s"${dep.username}/${dep.projectName}"
+      out.println(s"Installing $depName")
+      install(depName, Some(dep.version), path, null) match {
+        case Ok(_) => out.println(s"Installation of $depName completed")
+        case Err(e) => out.println(s"Installation of $depName failed"); return Err(e)
+      }
+    }
+    ().toOk
+  }
+
   /**
-    * Installs a flix package from the Github `project`.
+    *  Installs a flix package from the Github `project`.
     *
     * `project` must be of the form `<owner>/<repo>`
     *
     * The package is installed at `lib/<owner>/<repo>`
     */
-  def install(project: String, p: Path, o: Options): Result[Unit, Int] = {
+  //TODO: delete options?
+  def install(project: String, version: Option[SemVer], p: Path, o: Options)(implicit out: PrintStream): Result[Unit, PackageError] = {
     val proj = GitHub.parseProject(project)
-    val release = GitHub.getLatestRelease(proj)
+    val release = (version match {
+      case None => GitHub.getLatestRelease(proj)
+      case Some(ver) => GitHub.getSpecificRelease(proj, ver)
+    }) match {
+      case Ok(release) => release
+      case Err(e) => return Err(e)
+    }
+
     val assets = release.assets.filter(_.name.endsWith(".fpkg"))
     val lib = getLibraryDirectory(p)
-    val assetFolder = lib.resolve(proj.owner).resolve(proj.repo)
+    val repoFolder = lib.resolve(proj.owner).resolve(proj.repo)
+    val assetFolder = repoFolder.resolve(s"ver${release.version.toString()}")
+
+    if(!Files.isDirectory(assetFolder)) {
+      // clear the other versions from the folder
+      deleteDirectory(repoFolder.toFile)
```

Comment:
I don't think -- at the moment -- we should be deleting files.

Reply:
What I intend is that we check in `lib/`. If the library has not yet been downloaded, then we download it. Simple as that :)

Reply:
Sorry if I said something different earlier.

Reply:
I might just have misunderstood. But I will take the code to delete other versions out again

Reply:
OK, perhaps we can sync up in person sometime during the week?

Reply:
Yes - I am here today until ~15:15 and also tomorrow afternoon